code/game/machinery/autolathe.dm

```diff
@@ -78,63 +78,118 @@
 	if(panel_open)
 		wires.Interact(user)
 	else
-		ui_interact(user)
+		tgui_interact(user)
 
-/obj/machinery/autolathe/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+/obj/machinery/autolathe/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
 	if(!ui)
-		ui = new(user, src, ui_key, "autolathe.tmpl", name, 800, 550)
+		ui = new(user, src, ui_key, "Autolathe", name, 750, 700, master_ui, state)
 		ui.open()
 
-/obj/machinery/autolathe/ui_data(mob/user, ui_key = "main", datum/topic_state/state = GLOB.default_state)
+
+/obj/machinery/autolathe/tgui_static_data(mob/user)
+	var/list/data = list()
+	data["categories"] = categories
+	var/list/recipes = list()
+	for(var/v in files.known_designs)
+		var/datum/design/D = files.known_designs[v]
+		var/list/cost_list = design_cost_data(D)
+		var/list/matreq = list()
+		for(var/list/x in cost_list)
+			if(!x["amount"])
+				continue
+			if(x["name"] == "metal") // Do not use MAT_METAL or MAT_GLASS here.
+				matreq["metal"] = x["amount"]
+			if(x["name"] == "glass")
+				matreq["glass"] = x["amount"]
+		var/obj/item/I = D.build_path
+		var/maxmult = 1
+		if(ispath(D.build_path, /obj/item/stack))
+			maxmult = D.maxstack
+		recipes.Add(list(list(
+			"name" = D.name,
+			"category" = D.category,
+			"uid" = D.UID(),
+			"requirements" =  matreq,
+			"hacked" = ("hacked" in D.category) ? TRUE : FALSE,
+			"max_multiplier" = maxmult,
+			"image" = "[icon2base64(icon(initial(I.icon), initial(I.icon_state), SOUTH, 1))]"
+		)))
```

Comment:
I would like to see profiler results for this, as I feel like `icon2base64` may be expensive.

Reply:
Marking resolved because you stated in design chat that having TM'ed this and profiled it, the CPU is fine.